### ISSUE TITLE
 error=“ACL not found: auth method “consul-k8s-component-auth-method” not found

### ISSUE THREAD

**Issue by itsbibeksaini** [hashicorp/consul-k8s#1388]
## Overview:

I am doing fresh install for consul using helm. Following are the config used:

```yaml
global:
  name: consul
  enabled: true
  datacenter: testDatacenter
  gossipEncryption:
    autoGenerate: true
  tls:
    enabled: true
    enableAutoEncrypt: true
    verify: true
  acls:
    manageSystemACLs: true
server:
  replicas: 1
  securityContext:
    runAsNonRoot: false
    runAsUser: 0
connectInject:
  enabled: true
controller:
  enabled: true
```

however,  consul-server shows auth method “consul-k8s-component-auth-method” not found in logs

Here's complete consul-server-0 logs

### Logs
 ==> Starting Consul agent...
           Version: '1.12.3'
           Node ID: '4dae343b-07a6-6bbd-7ae3-35266b7f6995'
         Node name: 'consul-server-0'
        Datacenter: 'tmsolution' (Segment: '<all>')
            Server: true (Bootstrap: true)
       Client Addr: [0.0.0.0] (HTTP: -1, HTTPS: 8501, gRPC: -1, DNS: 8600)
      Cluster Addr: 10.1.0.12 (LAN: 8301, WAN: 8302)
           Encrypt: Gossip: true, TLS-Outgoing: true, TLS-Incoming: true, Auto-Encrypt-TLS: true

==> Log data will now stream in as it occurs:

2022-07-31T07:49:05.696Z [WARN]  agent: The 'ca_file' field is deprecated. Use the 'tls.defaults.ca_file' field instead.
2022-07-31T07:49:05.696Z [WARN]  agent: The 'cert_file' field is deprecated. Use the 'tls.defaults.cert_file' field instead.
2022-07-31T07:49:05.696Z [WARN]  agent: The 'key_file' field is deprecated. Use the 'tls.defaults.key_file' field instead.
2022-07-31T07:49:05.696Z [WARN]  agent: The 'verify_incoming_rpc' field is deprecated. Use the 'tls.internal_rpc.verify_incoming' field instead.
2022-07-31T07:49:05.696Z [WARN]  agent: The 'verify_outgoing' field is deprecated. Use the 'tls.defaults.verify_outgoing' field instead.
2022-07-31T07:49:05.696Z [WARN]  agent: The 'verify_server_hostname' field is deprecated. Use the 'tls.internal_rpc.verify_server_hostname' field instead.
2022-07-31T07:49:05.696Z [WARN]  agent: BootstrapExpect is set to 1; this is the same as Bootstrap mode.
2022-07-31T07:49:05.696Z [WARN]  agent: bootstrap = true: do not enable unless necessary
2022-07-31T07:49:06.106Z [WARN]  agent.auto_config: The 'ca_file' field is deprecated. Use the 'tls.defaults.ca_file' field instead.
2022-07-31T07:49:06.106Z [WARN]  agent.auto_config: The 'cert_file' field is deprecated. Use the 'tls.defaults.cert_file' field instead.
2022-07-31T07:49:06.106Z [WARN]  agent.auto_config: The 'key_file' field is deprecated. Use the 'tls.defaults.key_file' field instead.
2022-07-31T07:49:06.106Z [WARN]  agent.auto_config: The 'verify_incoming_rpc' field is deprecated. Use the 'tls.internal_rpc.verify_incoming' field instead.
2022-07-31T07:49:06.106Z [WARN]  agent.auto_config: The 'verify_outgoing' field is deprecated. Use the 'tls.defaults.verify_outgoing' field instead.
2022-07-31T07:49:06.106Z [WARN]  agent.auto_config: The 'verify_server_hostname' field is deprecated. Use the 'tls.internal_rpc.verify_server_hostname' field instead.
2022-07-31T07:49:06.106Z [WARN]  agent.auto_config: BootstrapExpect is set to 1; this is the same as Bootstrap mode.
2022-07-31T07:49:06.106Z [WARN]  agent.auto_config: bootstrap = true: do not enable unless necessary
2022-07-31T07:49:06.393Z [INFO]  agent.server.raft: initial configuration: index=1 servers="[{Suffrage:Voter ID:4dae343b-07a6-6bbd-7ae3-35266b7f6995 Address:10.1.0.12:8300}]"
2022-07-31T07:49:06.397Z [INFO]  agent.server.raft: entering follower state: follower="Node at 10.1.0.12:8300 [Follower]" leader=
2022-07-31T07:49:06.494Z [INFO]  agent.server.serf.wan: serf: EventMemberJoin: consul-server-0.tmsolution 10.1.0.12
2022-07-31T07:49:06.497Z [INFO]  agent.server.serf.lan: serf: EventMemberJoin: consul-server-0 10.1.0.12
2022-07-31T07:49:06.497Z [INFO]  agent.router: Initializing LAN area manager
2022-07-31T07:49:06.499Z [INFO]  agent.server.autopilot: reconciliation now disabled
2022-07-31T07:49:06.593Z [INFO]  agent.server: Handled event for server in area: event=member-join server=consul-server-0.tmsolution area=wan
2022-07-31T07:49:06.593Z [INFO]  agent.server: Adding LAN server: server="consul-server-0 (Addr: tcp/10.1.0.12:8300) (DC: tmsolution)"
2022-07-31T07:49:06.593Z [WARN]  agent: [core]grpc: addrConn.createTransport failed to connect to {tmsolution-10.1.0.12:8300 consul-server-0 <nil> 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial tcp <nil>->10.1.0.12:8300: operation was canceled". Reconnecting...
2022-07-31T07:49:06.793Z [INFO]  agent: Started DNS server: address=0.0.0.0:8600 network=tcp
2022-07-31T07:49:06.794Z [INFO]  agent: Started DNS server: address=0.0.0.0:8600 network=udp
2022-07-31T07:49:06.796Z [INFO]  agent: Starting server: address=[::]:8501 network=tcp protocol=https
2022-07-31T07:49:06.797Z [INFO]  agent: started state syncer
2022-07-31T07:49:06.797Z [INFO]  agent: Consul agent running!
2022-07-31T07:49:06.800Z [INFO]  agent: Retry join is supported for the following discovery methods: cluster=LAN discovery_methods="aliyun aws azure digitalocean gce k8s linode mdns os packet scaleway softlayer tencentcloud triton vsphere"
2022-07-31T07:49:06.800Z [INFO]  agent: Joining cluster...: cluster=LAN
2022-07-31T07:49:06.800Z [INFO]  agent: (LAN) joining: lan_addresses=[consul-server.default.svc:8301]
2022-07-31T07:49:06.909Z [WARN]  agent.server.memberlist.lan: memberlist: Failed to resolve consul-server.default.svc:8301: lookup consul-server.default.svc on 10.96.0.10:53: no such host
2022-07-31T07:49:06.909Z [WARN]  agent: (LAN) couldn't join: number_of_nodes=0 error="1 error occurred:
	* Failed to resolve consul-server.default.svc:8301: lookup consul-server.default.svc on 10.96.0.10:53: no such host

"
2022-07-31T07:49:06.909Z [WARN]  agent: Join cluster failed, will retry: cluster=LAN retry_interval=30s error=<nil>
2022-07-31T07:49:13.387Z [WARN]  agent.server.raft: heartbeat timeout reached, starting election: last-leader=
2022-07-31T07:49:13.387Z [INFO]  agent.server.raft: entering candidate state: node="Node at 10.1.0.12:8300 [Candidate]" term=2
2022-07-31T07:49:13.393Z [INFO]  agent.server.raft: election won: tally=1
2022-07-31T07:49:13.393Z [INFO]  agent.server.raft: entering leader state: leader="Node at 10.1.0.12:8300 [Leader]"
2022-07-31T07:49:13.394Z [INFO]  agent.server: cluster leadership acquired
2022-07-31T07:49:13.394Z [INFO]  agent.server: New leader elected: payload=consul-server-0
2022-07-31T07:49:13.397Z [INFO]  agent.server: initializing acls
2022-07-31T07:49:13.404Z [INFO]  agent.server: Created ACL 'global-management' policy
2022-07-31T07:49:13.407Z [WARN]  agent.server.acl: failed to remove bootstrap file: error="remove /consul/data/acl-bootstrap-reset: no such file or directory"
2022-07-31T07:49:13.493Z [INFO]  agent.server: Created ACL anonymous token from configuration
2022-07-31T07:49:13.493Z [INFO]  agent.leader: started routine: routine="legacy ACL token upgrade"
2022-07-31T07:49:13.493Z [INFO]  agent.leader: started routine: routine="acl token reaping"
2022-07-31T07:49:13.495Z [INFO]  agent.leader: stopping routine: routine="legacy ACL token upgrade"
2022-07-31T07:49:13.495Z [INFO]  agent.leader: stopped routine: routine="legacy ACL token upgrade"
2022-07-31T07:49:13.520Z [WARN]  agent: Node info update blocked by ACLs: node=4dae343b-07a6-6bbd-7ae3-35266b7f6995 accessorID=00000000-0000-0000-0000-000000000002
2022-07-31T07:49:13.595Z [INFO]  agent.server.acl: ACL bootstrap completed
2022-07-31T07:49:13.599Z [INFO]  agent.server.autopilot: reconciliation now enabled
2022-07-31T07:49:13.599Z [INFO]  agent.leader: started routine: routine="federation state anti-entropy"
2022-07-31T07:49:13.599Z [INFO]  agent.leader: started routine: routine="federation state pruning"
2022-07-31T07:49:13.702Z [INFO]  connect.ca: initialized primary datacenter CA with provider: provider=consul
2022-07-31T07:49:13.702Z [INFO]  agent.leader: started routine: routine="intermediate cert renew watch"
2022-07-31T07:49:13.702Z [INFO]  agent.leader: started routine: routine="CA root pruning"
2022-07-31T07:49:13.702Z [INFO]  agent.leader: started routine: routine="CA root expiration metric"
2022-07-31T07:49:13.702Z [INFO]  agent.leader: started routine: routine="CA signing expiration metric"
2022-07-31T07:49:13.702Z [INFO]  agent.leader: started routine: routine="virtual IP version check"
2022-07-31T07:49:13.708Z [INFO]  agent.server: member joined, marking health alive: member=consul-server-0 partition=default
2022-07-31T07:49:13.712Z [INFO]  agent.leader: stopping routine: routine="virtual IP version check"
2022-07-31T07:49:13.792Z [INFO]  agent.leader: stopped routine: routine="virtual IP version check"
2022-07-31T07:49:13.984Z [INFO]  agent.server: federation state anti-entropy synced
2022-07-31T07:49:15.341Z [WARN]  agent: Node info update blocked by ACLs: node=4dae343b-07a6-6bbd-7ae3-35266b7f6995 accessorID=00000000-0000-0000-0000-000000000002
2022-07-31T07:49:17.103Z [INFO]  agent: Updated agent's ACL token: token=agent
2022-07-31T07:49:19.126Z [INFO]  agent: Synced node info
2022-07-31T07:49:24.688Z [ERROR] agent.http: Request error: method=PUT url=/v1/acl/policy from=10.1.0.16:39682 error="Invalid Policy: A Policy with Name "agent-token" already exists"
2022-07-31T07:49:24.778Z [INFO]  agent: Updated agent's ACL token: token=agent
2022-07-31T07:49:25.610Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:26.613Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:27.615Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:28.617Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:29.621Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:30.637Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:31.669Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:32.673Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:33.571Z [ERROR] agent.http: Request error: method=PUT url=/v1/acl/policy from=10.1.0.17:40394 error="Invalid Policy: A Policy with Name "agent-token" already exists"
2022-07-31T07:49:33.672Z [INFO]  agent: Updated agent's ACL token: token=agent
2022-07-31T07:49:33.677Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:34.680Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:35.685Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:36.688Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:36.886Z [INFO]  agent: (LAN) joining: lan_addresses=[consul-server.default.svc:8301]
2022-07-31T07:49:36.970Z [INFO]  agent: (LAN) joined: number_of_nodes=1
2022-07-31T07:49:36.970Z [INFO]  agent: Join cluster completed. Synced with initial agents: cluster=LAN num_agents=1
2022-07-31T07:49:37.691Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:38.696Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:39.700Z [ERROR] agent.http: Request error: method=POST url=/v1/acl/login from=10.1.0.8:50920 error="ACL not found: auth method "consul-k8s-component-auth-method" not found"
2022-07-31T07:49:40.273Z [ERROR] agent.http: Request error: method=PUT url=/v1/acl/policy from=10.1.0.18:35220 error="Invalid Policy: A Policy with Name "agent-token" already exists" 

am I missing something in config?

**Comment by itsbibeksaini** [comment]
duplicate of #1145

**Comment by david-yu** [comment]
Thanks @itsbibeksaini we are looking to resolve this sometime after our upcoming release happening later this month.
